shared/go/pkg/valueobject/email_address.go

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright © 2021 the contributors.
+ *
+ *  This file is part of the DaSCH Service Platform.
+ *
+ *  The DaSCH Service Platform is free software: you can
+ *  redistribute it and/or modify it under the terms of the
+ *  GNU Affero General Public License as published by the
+ *  Free Software Foundation, either version 3 of the License,
+ *  or (at your option) any later version.
+ *
+ *  The DaSCH Service Platform is distributed in the hope that
+ *  it will be useful, but WITHOUT ANY WARRANTY; without even
+ *  the implied warranty of MERCHANTABILITY or FITNESS FOR
+ *  A PARTICULAR PURPOSE.  See the GNU Affero General Public
+ *  License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public
+ *  License along with the DaSCH Service Platform.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
```

Comment:
Seriously we need to have it in every single file?

Reply:
I've been wondering about this too.

Reply:
no idea. we had it in dsp-api in every source file. I have it as a template in IntelliJ which gets automatically added to each file I create.

According to this, unfortunately, yes: https://softwareengineering.stackexchange.com/questions/317041/should-i-add-the-license-in-every-header-and-source-file/357796

Reply:
According to the article, it would be just as valid, if we had one line `DaSCH 2021, AGPL`, or am I mistaken?

... but good to know for my python stuff.

Reply:
It's just on of many answers for this question, but if you check the others, [they are divided](https://softwareengineering.stackexchange.com/questions/125836/do-you-have-to-include-a-license-notice-with-every-source-file).

I think this is kind of old approach to have it everywhere. At least for me it doesn't make sense, especially when all files are under the same license, therefore I would say we should only one `LICENSE.md` file in the root folder. However if part of repository should have different licence type, then this should be highlighted.

Reply:
Yes, a lot of opinions. In this case it is more helpful to just go to the source: https://www.gnu.org/licenses/gpl-howto.html

Reply:
As far as I understand it, it needs to be in each file, if we want to do it correctly, which I think we should. What I'm not so sure about, is, if we are actually allowed to publish this software with a GPL license. But this is out of scope for this discussion.